src/QueryBuilder.jsx

```diff
@@ -144,11 +146,11 @@ export default class QueryBuilder extends React.Component {
     }
 
     componentWillMount() {
-        const {fields, operators, combinators, controlElements, controlClassnames} = this.props;
+        const {fields, initValue, operators, combinators, controlElements, controlClassnames} = this.props;
```

Comment:
You are thinking right! I suggest we use the `this.props.query` itself to take in the initial value. There would be no need for a new prop.